client/ayon_core/pipeline/workfile/path_resolving.py

```diff
@@ -207,12 +214,94 @@ def get_workdir(
     )
 
 
-def get_last_workfile_with_version(
-    workdir, file_template, fill_data, extensions
-):
+@dataclass
+class WorkfileParsedData:
+    version: Optional[int] = None
+    comment: Optional[str] = None
+    ext: Optional[str] = None
+
+
+class WorkfileDataParser:
```

Comment:
Just wanted to mention that something along these lines does exist in the workfiles tool logic. Or at least, e.g. getting the "subversion" or "comment" from the filename.

It's around here: https://github.com/ynput/ayon-core/blob/b0ac87b7b12006dc8b3fffe441b91a884cc65ce5/client/ayon_core/tools/workfiles/models/workfiles.py#L360-L366
Which uses this [`CommentMatcher`](https://github.com/ynput/ayon-core/blob/b0ac87b7b12006dc8b3fffe441b91a884cc65ce5/client/ayon_core/tools/workfiles/models/workfiles.py#L31-L83).

I wonder if this should get aligned more and maybe have the code there use this? or? seems like a lot of overlap.

Reply:
> Which uses this CommentMatcher.

`CommentMatcher` is not used anymore, I've marked it as deprecated.